core/src/net/network_thread.rs

```diff
@@ -1009,6 +1010,18 @@ impl NetworkThread {
     }
 }
 
+impl std::ops::Drop for NetworkThread {
+    fn drop(&mut self) {
+        // Ensure that the channels are shutdown and buffers are deallocated on panic
+        if !self.stopped {
+            while let Some((_, channel)) = self.token_map.pop_lru() {
+                trace!(self.log, "Dropping channel in crashed NetworkThread");
+                self.drop_channel(&mut *channel.borrow_mut());
```

Comment:
Why the `channel` -> `&mut channel` -> `channel` -> `&mut channel` back and forth?

Reply:
It's `Rc<RefCell<TcpChannel>>` -> `RefMut<TcpChannel>` -> `TcpChannel` -> `&mut TcpChannel`

Reply:
Ah, so it should just be `channel.borrow_mut().deref_mut()`.